sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/Account.kt

```diff
@@ -1,16 +1,44 @@
 package cash.z.ecc.android.sdk.model
 
+import cash.z.ecc.android.sdk.internal.jni.JNI_ACCOUNT_UUID_BYTES_SIZE
+
 /**
- * A [ZIP 32](https://zips.z.cash/zip-0032) account index.
+ * Unique identifier for a specific account tracked by a `Synchronizer`.
+ *
+ * Account identifiers are "one-way stable": a given identifier always points to a
+ * specific viewing key within a specific `Synchronizer` instance, but the same viewing
+ * key may have multiple account identifiers over time. In particular, this SDK upholds
+ * the following properties:
+ *
+ * - When an account starts being tracked within a `Synchronizer` instance, a new
+ *   `Account` is generated.
+ * - If an `Account` is present within a `Synchronizer`, it always points to the same
+ *   account.
+ *
+ * What this means is that account identifiers are not stable across "wallet recreation
+ * events". Examples of these include:
+ * - Restoring a wallet from a backed-up seed.
+ * - Importing the same viewing key into two different wallet instances.
  *
- * @param value A 0-based account index.  Must be >= 0.
+ * @param accountUuid The account identifier. Must be length 16.
  */
-data class Account(val value: Int) {
+data class Account(val accountUuid: ByteArray) {
```

Comment:
Why not call this `AccountUuid` and use it as the type passed to identify an account in other APIs (rather than `ByteArray`)? That would eliminate the need to explicitly require it to be 16 bytes in most places.

Reply:
Yes, we can do this optimization.

Reply:
Done